File: src/sugtao4423/lod/userpage_fragment/UserPage.java
package sugtao4423.lod.userpage_fragment;

import twitter4j.TwitterException;
import twitter4j.User;

import android.graphics.Color;
import android.os.AsyncTask;
import android.os.Bundle;
import android.support.v4.app.FragmentActivity;
import android.support.v4.view.PagerTabStrip;
import android.support.v4.view.ViewPager;
import sugtao4423.lod.ApplicationClass;
import sugtao4423.lod.R;
import sugtao4423.lod.ShowToast;

public class UserPage extends FragmentActivity{

	public static final String INTENT_EXTRA_KEY_USER_OBJECT = "userObject";
	public static final String INTENT_EXTRA_KEY_USER_SCREEN_NAME = "userScreenName";

	private User target;
	private UserPageFragmentPagerAdapter adapter;
	private ApplicationClass appClass;

	@Override
	protected void onCreate(Bundle savedInstanceState){
		super.onCreate(savedInstanceState);
		setContentView(R.layout.userpage);

		adapter = new UserPageFragmentPagerAdapter(getSupportFragmentManager());
		ViewPager viewPager = (ViewPager)findViewById(R.id.Userpager);
		viewPager.setAdapter(adapter);
		viewPager.setOffscreenPageLimit(5);

		PagerTabStrip strip = (PagerTabStrip)findViewById(R.id.userPagerTabStrip);
		strip.setTabIndicatorColor(Color.parseColor(getString(R.color.pagerTabIndicator)));
		strip.setDrawFullUnderline(true);
		getActionBar().setDisplayShowHomeEnabled(false);

		appClass = (ApplicationClass)UserPage.this.getApplicationContext();

		target = (User)getIntent().getSerializableExtra(INTENT_EXTRA_KEY_USER_OBJECT);
		if(target != null){
			setTargetUser();
			return;
		}

		new AsyncTask<Void, Void, User>(){
			@Override
			protected User doInBackground(Void... params){
				try{
					return appClass.getTwitter().showUser(getIntent().getStringExtra(INTENT_EXTRA_KEY_USER_SCREEN_NAME));
				}catch(TwitterException e){
					return null;
				}
			}

			@Override
			protected void onPostExecute(User result){
				if(result != null){
					target = result;
					setTargetUser();
					((_0_detail)(adapter.getItem(0))).setText();
				}else{
					new ShowToast(R.string.error_getUserDetail, UserPage.this, 0);
					finish();
				}
			}
		}.execute();
	}

	public void setTargetUser(){
		getActionBar().setTitle(target.getName());
		((_0_detail)(adapter.getItem(0))).setTargetUser(target);
		((_1_Tweet)(adapter.getItem(1))).setTargetUser(target);
		((_2_favorites)(adapter.getItem(2))).setTargetUser(target);
		((_3_follow)(adapter.getItem(3))).setTargetUser(target);
		((_4_follower)(adapter.getItem(4))).setTargetUser(target);
	}

}

File: app/src/main/java/sugtao4423/lod/userpage_fragment/UserPage.kt
package sugtao4423.lod.userpage_fragment

import android.os.Bundle
import androidx.core.content.ContextCompat
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import kotlinx.coroutines.withContext
import sugtao4423.lod.LoDBaseActivity
import sugtao4423.lod.R
import sugtao4423.lod.ShowToast
import sugtao4423.lod.databinding.UserpageBinding
import twitter4j.TwitterException
import twitter4j.User

class UserPage : LoDBaseActivity() {

    companion object {
        const val INTENT_EXTRA_KEY_USER_OBJECT = "userObject"
        const val INTENT_EXTRA_KEY_USER_SCREEN_NAME = "userScreenName"
    }

    private lateinit var adapter: UserPageFragmentPagerAdapter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding = UserpageBinding.inflate(layoutInflater)
        setContentView(binding.root)

        adapter = UserPageFragmentPagerAdapter(supportFragmentManager, this)
        binding.userPager.let {
            it.adapter = adapter
            it.offscreenPageLimit = 5
        }

        binding.userPagerTabStrip.apply {
            tabIndicatorColor = ContextCompat.getColor(applicationContext, R.color.pagerTabText)
            drawFullUnderline = true
        }
        supportActionBar?.setDisplayShowHomeEnabled(false)

        intent.getSerializableExtra(INTENT_EXTRA_KEY_USER_OBJECT).let {
            if (it != null) {
                setTargetUser(it as User)
                return
            }
        }

        CoroutineScope(Dispatchers.Main).launch {
            val result = withContext(Dispatchers.IO) {
                try {
                    app.twitter.showUser(intent.getStringExtra(INTENT_EXTRA_KEY_USER_SCREEN_NAME))
                } catch (e: TwitterException) {
                    null
                }
            }
            if (result != null) {
                setTargetUser(result)
            } else {
                ShowToast(applicationContext, R.string.error_get_user_detail)
                finish()
            }
        }
    }

    private fun setTargetUser(target: User) {
        supportActionBar?.title = target.name
        adapter.setTargetUser(target)
    }

}
